### DISCUSSION TITLE
Single Tiff file from a PDF that have multiple pages

### DISCUSSION THREAD

**Discussion by VaibJo2002** [dlemstra/Magick.NET#1363] [Help]
I was going through the code and trying to run it and it is converting all the pages of pdf to individual tiff image and that each tiff image contains 7 parts which are same but gets smaller when i go from 1->7.
How do I implement this code to get single tiff file from multi page pdf?

**Comment by dlemstra** [comment]
Can you share a sample file so I can better understand what you are asking me?

**Reply by VaibJo2002** [reply]
https://drive.google.com/file/d/1PfuIM4C-R1VK2fKWKKdRy0W2dFc1MGPZ/view?usp=sharing
This is a sample test case file. I want to convert this pdf into single tiff images with multiple images. When I am using pdf to tiff code i am getting separate tiff file for each page in pdf.

**Reply by dlemstra** [reply]
How are you saving the output image?

**Reply by VaibJo2002** [reply]
in the local directory as tiff image

**Reply by dlemstra** [reply]
I should have been more clear. What is the code that you are using to write the image? I suspect that you are using the example from this repository? When you use `images.Write` instead of the `foreach` you should get the correct output image.

**Reply by VaibJo2002** [reply]
yes I'm using the code from the example. I am a fresher at job and I need to get this working soon.
to do: Convert a pdf file to tiff file.
Result from your pdf convert code -> Getting tiff files of each page but not a single one.
Pls help

**Reply by dlemstra** [reply]
Can you show a code sample of what you are doing? I suspect there is a mistake in your code but without seeing the code I cannot tell you where.

**Reply by VaibJo2002** [reply]
I was going through other discussions and i finally managed to did it.
here's my sample code ->
`````
class Program
    {
        static void Main(string[] args)
        {
            var settings = new MagickReadSettings();
            // Settings the density to 300 dpi will create an image with a better quality
            settings.Density = new Density(75);

            using (var images = new MagickImageCollection())
            {
                // Add all the pages of the pdf file to the collection
                images.Read(@"D:\Projects\Pdf to Tiff\PdfToTiff\Input\yes.pdf", settings);

                using (var tif = new MagickImageCollection())
                {
                    var page = 1;
                    foreach (var image in images)
                    {
                        image.Format = MagickFormat.Tiff;
                        tif.Add(image);
                        page++;
                    }
                    tif.Write(@"D:\Projects\Pdf to Tiff\PdfToTiff\Input" + ".tif", MagickFormat.Tiff);
                }
            }

        }
    }
`````

Now I have also written the API in c# that gives response in stream and the data goes into azure blob.
But I have hit a Major Roadblock. Our production in on only Azure not Azure VM. And I don't have the slightest idea that installing GhostScript in Azure is possible or not and make this api runnable. Your help is appreciated.

**Reply by dlemstra** [reply]
The documentation of this project explains what you should do to configure Ghostscript: https://github.com/dlemstra/Magick.NET/tree/main/docs#ghostscript. And I hope you are aware that you need a license if you want to use this in a commercial project.
